crates/store/re_dataframe2/src/query.rs

```diff
@@ -0,0 +1,644 @@
+                                    component: selected_component_name,
+                                    join_encoding: _
+                                } = selected_column;
+
+                                view_contents
+                                    .iter()
+                                    .filter_map(|view_column| match view_column {
+                                        ColumnDescriptor::Component(view_descr) => Some(view_descr),
+                                        _ => None,
+                                    })
+                                    .find(|view_descr| {
+                                        view_descr.entity_path == *selected_entity_path
+                                        && view_descr.component_name == *selected_component_name
+                                    })
+                                    .cloned()
+                                    .map_or_else(
+                                        || ColumnDescriptor::Component(ComponentColumnDescriptor {
+                                            entity_path: selected_entity_path.clone(),
+                                            archetype_name: None,
+                                            archetype_field_name: None,
+                                            component_name: *selected_component_name,
+                                            store_datatype: arrow2::datatypes::DataType::Null,
+                                            join_encoding: JoinEncoding::default(),
+                                            is_static: false,
+                                        }),
+                                        ColumnDescriptor::Component,
+                                    )
+                            },
+                        }
+                    }).collect_vec()
+            } else {
+                view_contents.clone()
+            };
+
+            // 3. Compute the Arrow schema of the selected components.
+            //
+            // Every result returned using this `QueryHandle` will match this schema exactly.
+            let arrow_schema =
+                ArrowSchema {
+                    fields: selected_contents
+                        .iter()
+                        .map(|descr| descr.to_arrow_field())
+                        .collect_vec(),
+                    metadata: Default::default(),
+                };
+
+            // 4. Perform the query and keep track of all the relevant chunks.
+            let chunks = {
+                let index_range = self
+                    .query
+                    .filtered_index_range
+                    .unwrap_or(ResolvedTimeRange::EVERYTHING);
+
+                let query = RangeQuery::new(self.query.filtered_index, index_range)
+                    .keep_extra_timelines(true) // we want all the timelines we can get!
+                    .keep_extra_components(false);
+
+                selected_contents
```

Comment:
I believe this should be `view_contents`; we need to look at all the chunks in the view to determine the timeline values so that rows are consistent regarless of selected columns.

That said, we don't actually need to sort / dedpuc the data-columns of those chunks; only the time columns.